### PR TITLE
ebpf: increase number of stack delta buckets

### PR DESCRIPTION
**Pull request by florianl**
With https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/230 an issue is reported, where the number of stack delta buckets is not sufficient.
Increase stack delta bucket to be able to unwind also such executables.

Fixes https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/230.

### CODE REVIEW

File: processmanager/ebpf/ebpf_test.go [hunk 1]
```diff
@@ -38,6 +38,6 @@ func TestMapID(t *testing.T) {
 		})
 	}
 
-	_, err := getMapID(1 << 22)
+	_, err := getMapID(1 << 24)
```
Comment (rockdaboot) [inline processmanager/ebpf/ebpf_test.go:41]:
```suggestion
	_, err := getMapID(1 << support.StackDeltaBucketLargest)
```

File: processmanager/ebpf/ebpf_test.go [hunk 2]
```diff
@@ -38,6 +38,6 @@ func TestMapID(t *testing.T) {
 		})
```
Comment (rockdaboot) [inline processmanager/ebpf/ebpf_test.go:38]: Suggestion: Add another testcase above:
```
		(1 << support.StackDeltaBucketLargest) - 1: support.StackDeltaBucketLargest,
```

**Comment by danipozo** [comment]
Missing this (I think) in `tools/coredump/ebpfhelpers.go`:
```diff
--        &C.exe_id_to_20_stack_deltas, &C.exe_id_to_21_stack_deltas:
++        &C.exe_id_to_20_stack_deltas, &C.exe_id_to_21_stack_deltas, &C.exe_id_to_22_stack_deltas,
++        &C.exe_id_to_23_stack_deltas:
```

**Comment by florianl** [comment]
> Missing this (I think) in tools/coredump/ebpfhelpers.go:

Good catch :+1: updated the proposed change.

File: processmanager/ebpf/ebpf_test.go (hunk 2 above)
Comment (rockdaboot) [reply processmanager/ebpf/ebpf_test.go:39]: To clarify, it covers the corner case of the highest acceptable value, which was hard-coded previously.

**Review by rockdaboot — APPROVED** [review]

**Review by felixge — APPROVED** [review]

**Review by athre0z — APPROVED** [review]

**Comment by rockdaboot** [comment]
Sorry @florianl, I accidentally pushed with `--force-with-lease` 🫨.

**Pull request merged**